### DISCUSSION TITLE
Error, can't find/create omnidata.h5

### DISCUSSION THREAD

**Discussion by runyiliu** [spacepy/spacepy#616] [Support and Q&A] [answered]
I am writing for help in using SpacePy(I am a new user of it). I found there is always an error "OSError: Unable to open file (unable to open file: name = '/root/.spacepy/data/omnidata.h5', errno = 2, error message = 'No such file or directory', flags = 0, o_flags = 0)" in running the code and there is no 'omnidata.h5' in the corresponding path indeed. I tried to solve it by using "spacepy.toolbox.update(all=True)" but it always resulted in "Connection timed out". The errors reported from both colab and my own PC code runner are totally the same.

What's more, as Dr. Morley suggested, I tried running "spacepy.toolbox.update(QDomni=True)", but still resulted in "Connection timed out". And I believe I am running it as root user.

The screenshots of running interface are attached below(in colab). I tried it in my own PC and just got same.
![1](https://user-images.githubusercontent.com/98632297/174916821-efa26122-d7c8-493d-8c2a-d2c935439053.png)
![2](https://user-images.githubusercontent.com/98632297/174916837-ada9d68d-23a1-4e92-bf35-e5f703a9740d.png)
![3](https://user-images.githubusercontent.com/98632297/174916844-6cba16ce-17ba-4f56-8706-32d9dc6f2719.png)
.

**Comment by spacecataz** [comment]
Hello!

I can reproduce this on my machine. However, the critical issue, the HTML request returning a time-out error, is not a code issue; it's that the website that hosts the Qin Denton data is down and not responding to the request. We're looking into that now and will get back soon.

**Comment by jtniehof** [comment, marked as answer]
In the short term, I have hosted the output HDF5 files on the [PSP/IS☉IS SOC](https://spp-isois.sr.unh.edu/data_public/omni/). They can be placed directly in the ``data`` subdirectory of [the .spacepy directory](https://spacepy.github.io/configuration.html).

When the OMNI data (the relevant h5 files) are not found, the import of omni should issue a warning:
```python
import spacepy.omni
/home/jtniehof/.local//lib//python3.6/site-packages/spacepy/omni.py:419: UserWarning: Qin-Denton/OMNI2 data not found in current format. This module has limited functionality. Run spacepy.toolbox.update(QDomni=True) to download data.
  "Qin-Denton/OMNI2 data not found in current format."
```

The Qin-Denton files are provided by ViRBO (for the bulk of the archive) and NMC (for latest up-to-date). Both servers are currently down, so the update cannot be pulled. We are currently planning long-term reliable sources for these files.

The OMNI2 data are normally pulled from SPDF and can be updated:
```python
import spacepy.toolbox
spacepy.toolbox.update(omni2=True)
```

It's possible to then draw from the OMNI 2 database in most cases
```python
import spacepy.omni
import spacepy.time

t = spacepy.time.Ticktock(['2002-02-02T12:00:00', '2002-02-02T12:10:00'], 'ISO')
omnivals = spacepy.omni.get_omni(t, dbase='OMNI2hourly')
```

However, the irbem functions don't accept OMNI2 values as inputs (#617), so this doesn't help out immediately.

Also, unless there is a specific reason, it's generally a poor idea to run as root.

**Reply by jtniehof** [reply]
See #628 if using this file gives very unusual times. (Short answer: update SpacePy).

**Reply by jtniehof** [reply]
The ViRBO and NMC sources are back up, so `toolbox.update()` should work.

**Reply by CptMoroni** [reply]
It looks like VIRBO is down again, at least I can't pull it up using a web browser and toolbox.update() is crashing.  I went to your [PSP/IS☉IS SOC](https://spp-isois.sr.unh.edu/data_public/omni/) link and downloaded  what you have there.  Thanks!

My question is, where is the NMC website and what do you mean by NMC has the latest up-to date files?  It looks like your last update to the omnidata.h5 was May 20, 2024, would NMC have a more up-to-date file?  And how would I append it to the omnidata.h5 file you provided if it did?

Thanks!

**Reply by jtniehof** [reply]
The NMC RBSP-ECT server [hosted the Q-D files](https://rbsp-ect.newmexicoconsortium.org/data_pub/QinDenton/) but that's no longer being updated since we're well past end of mission. We're working to get them hosted regularly with SPDF but that process is dragging out. #731

**Comment by runyiliu** [comment]
Thank you so much! @jtniehof
